File: ruby/data structures/Stack.rb
##########################################
## Stack Data Structure
##########################################
## 6-30-2020
##########################################

class Stack
    def initialize
        @data = []
    end

    def push (data)
        return @data.push data
    end

    def pop
        return @data.pop
    end

    def getStack
        return @data
    end
end

########Example Code##########


PileOfPapers = Stack.new

PileOfPapers.push 1
PileOfPapers.push 2
PileOfPapers.push 3
puts PileOfPapers.getStack
PileOfPapers.pop
puts PileOfPapers.getStack

File: js/Data Structures/stack.js
//////////////////////////////
// Create a Stack Structure
// 6/28/20
// Should be able to push data on top of stack
// Pop data from top of stack
//////////////////////////////

////////////CLASS/////////////////////

class Stack {
    #data = [];

    constructor(data) {
        if (data instanceof Array) {
            this.#data = [...data]; //can pass initial array for initial data
        }
        //Data is setup as private class field (#) so users can't access inner array directly, only through class methods
    }

    push(value) {
        this.#data.push(value); //add data to end of data
    }

    pop() {
        this.#data.pop(); //remove last item from data
    }

    inspect() {
        return this.#data; //getter function for data
    }
}

//////////////////////////////////////

/////////TEST CODE/////////////////////

const myStack = new Stack([1, 2, 3, 4, 5]);

console.log(myStack.inspect()); //Show the initial stack of the stack

myStack.pop(); //remove value from stack
myStack.push(6); // add value to stack

console.log(myStack.inspect()); //show final state of Stack

const emptyStack = new Stack(6); //show what happens if pass in non-array value

console.log(emptyStack.inspect());

///////////////////////////////////////
